File: pp.cpp
#include <iostream>
using namespace std;

class Music
{
protected:
    string mood;

public:
    void setMood(string m)
    {
        mood = m;
    }

    virtual void recommend() = 0;   
};

class Song : public Music
{
private:
    string name;
    string artist;

public:
  
    Song()
    {
        name = "Unknown";
        artist = "Unknown";
    }

    Song(string n, string a)
    {
        name = n;
        artist = a;
    }

    void recommend()
    {
        cout << "\nRecommended Song: " << name << " by " << artist << endl;
    }

    void setDetails(string n, string a)
    {
        name = n;
        artist = a;
    }

    void getDetails()
    {
        cout << "Song Name : " << name << endl;
        cout << "Artist    : " << artist << endl;
    }

    ~Song()
    {
        cout << "\nThank You for using Song Recommender!\n";
    }
};

int main()
{
    string userMood;

    cout << "Enter Mood (happy/sad/romantic/motivational/chill): ";
    cin >> userMood;

    Song s1;   

    if (userMood == "happy")
        s1.setDetails("Cruel Summer", "Taylor Swift");

    else if (userMood == "sad")
        s1.setDetails("Let Me Down Slowly", "Alec Benjamin");

    else if (userMood == "romantic")
        s1.setDetails("Perfect", "Ed Sheeran");

    else if (userMood == "motivational")
        s1.setDetails("Unstoppable", "Sia");

    else if (userMood == "chill")
        s1.setDetails("Levitating", "Dua Lipa");

    else
    {
        cout << "Invalid Mood!" << endl;
        return 0;
    }

    s1.setMood(userMood);

    Music *ptr;
    ptr = &s1;

    ptr->recommend();

    cout << endl;
    s1.getDetails();

    return 0;
}

File: pp.js
function recommendSong() {

    let mood = document.getElementById("mood").value;
    let result = document.getElementById("result");

    let songLink = "";

    if (mood == "happy") {
        result.innerHTML = "🎵 Cruel Summer by Taylor Swift";
        songLink = "https://open.spotify.com/track/1BxfuPKGuaTgP7aM0Bbdwr?si=b8e14dc81f1b4f6f";
    }

    else if (mood == "sad") {
        result.innerHTML = "🎵 Let Me Down Slowly by Alec Benjamin";
        songLink = "https://open.spotify.com/track/2qxmye6gAegTMjLKEBoR3d?si=9df287f4e4f34cb9";
    }

    else if (mood == "romantic") {
        result.innerHTML = "🎵 Perfect - Ed Sheeran";
        songLink = "https://open.spotify.com/track/0tgVpDi06FyKpA1z0VMD4v?si=430b115de7284a92";
    }

    else if (mood == "motivational") {
        result.innerHTML = "🎵 Unstopable by Sia";
        songLink = "https://open.spotify.com/track/2J2Z1SkXYghSajLibnQHOa?si=0e63f9f90d8e4118";
    }

    else if (mood == "chill") {
        result.innerHTML = "🎵 Levitating by Dua Lipa";
        songLink = "https://open.spotify.com/track/39LLxExYz6ewLAcYrzQQyP?si=933afa5ae0e04455";
    }

    window.open(songLink, "_blank");
}